database/interface.py

```diff
@@ -419,57 +419,90 @@ def update_harvest_record(self, record_id, updates):
             self.db.rollback()
             return None
 
-    def delete_harvest_record(self, identifier):
-        records = self.db.query(HarvestRecord).filter_by(identifier=identifier).all()
-        if not records:
-            logger.warning(f"Harvest records with identifier {identifier} not found")
-            return
-        logger.info(
-            f"{len(records)} records with idenitifier {identifier}\
+    def delete_harvest_record(self, identifier=None, ID=None, harvest_source_id=None):
```

Comment:
minor gripe, but can we use record_id here instead of ID? feels a little more pythonic.